storybooknative/src/Button/index.js

```diff
@@ -1,13 +1,171 @@
+      [TYPE_OPTIONS.SUCCESS]: theme.orbit.colorTextButtonSuccess,
+      [TYPE_OPTIONS.WARNING]: theme.orbit.colorTextButtonWarning,
+      [TYPE_OPTIONS.CRITICAL]: theme.orbit.colorTextButtonCritical,
+      [TYPE_OPTIONS.FACEBOOK]: theme.orbit.colorTextButtonFacebook,
+      [TYPE_OPTIONS.GOOGLE]: theme.orbit.colorTextButtonGoogle,
+    },
+    [TOKENS.colorTextButtonBordered]: {
+      [TYPE_OPTIONS.PRIMARY]: theme.orbit.colorTextButtonPrimaryBordered,
+      [TYPE_OPTIONS.SECONDARY]: theme.orbit.colorTextButtonSecondaryBordered,
+      [TYPE_OPTIONS.INFO]: theme.orbit.colorTextButtonInfoBordered,
+      [TYPE_OPTIONS.SUCCESS]: theme.orbit.colorTextButtonSuccessBordered,
+      [TYPE_OPTIONS.WARNING]: theme.orbit.colorTextButtonWarningBordered,
+      [TYPE_OPTIONS.CRITICAL]: theme.orbit.colorTextButtonCriticalBordered,
+      [TYPE_OPTIONS.FACEBOOK]: theme.orbit.colorTextButtonFacebookBordered,
+      [TYPE_OPTIONS.GOOGLE]: theme.orbit.colorTextButtonGoogleBordered,
+    },
+  };
+  return tokens[name][type];
+};
+
+export const StyledButton = styled(View, props => {
+  const { bordered, theme, width, circled, disabled } = props;
+  return {
+    width,
+    justifyContent: "center",
+    alignItems: "center",
+    height: getSizeToken(TOKENS.heightButton, props),
+    backgroundColor: bordered
+      ? theme.orbit.backgroundButtonBordered
+      : getTypeToken(TOKENS.backgroundButton, props),
+    borderWidth: bordered ? "1px" : "0px",
+    borderColor: getTypeToken(TOKENS.borderColorButton, props),
+    borderRadius: circled
+      ? getSizeToken(TOKENS.heightButton, props)
+      : theme.orbit.borderRadiusNormal,
+    padding: 0,
+    paddingLeft: getSizeToken(TOKENS.paddingButton, props),
+    paddingRight: getSizeToken(TOKENS.paddingButton, props),
+    opacity: disabled && theme.orbit.opacityButtonDisabled,
+  };
+});
+
+StyledButton.defaultProps = {
+  theme: defaultTokens,
+};
+
+const StyledText = styled(Text, props => {
+  const { bordered, theme } = props;
+  return {
+    color: bordered
+      ? getTypeToken(TOKENS.colorTextButtonBordered, props)
+      : getTypeToken(TOKENS.colorTextButton, props),
+    fontWeight: theme.orbit.fontWeightBold,
+    fontSize: getSizeToken(TOKENS.fontSizeButton, props),
+  };
+});
+
+StyledText.defaultProps = {
+  theme: defaultTokens,
+};
+
+const StyledTouchableWrapper = styled(View, props => ({
+  overflow: "hidden",
+  width: props.width,
+  borderRadius: props.circled
+    ? getSizeToken(TOKENS.heightButton, props)
+    : props.theme.orbit.borderRadiusNormal,
+}));
+
+StyledTouchableWrapper.defaultProps = {
+  theme: defaultTokens,
+};
+
+const Button = (props: Props) => {
+  const {
+    onPress = () => {},
+    children,
+    size = SIZE_OPTIONS.NORMAL,
+    type = TYPE_OPTIONS.PRIMARY,
+    ...rest
+  } = props;
+  let width;
+  if (props.block) {
+    width = "100%";
+  } else {
+    width = props.width ? `${props.width}px` : "auto";
+  }
+
+  if (Platform.OS === "android") {
```

Comment:
Should this live in a different _platform-specific_ file? The button component will be used everywhere, so I don't know if that makes sense that this `if` statement is performed every time.

Reply:
Okay, let's say for now this is a possible improvement which does not need to be implemented yet.